src/blas/impl/KokkosBlas1_iamax_impl.hpp

```diff
@@ -86,10 +86,36 @@ struct V_Iamax_Functor
                    "RV must have rank 0 and XV must have rank 1.");
   }
 
-  KOKKOS_INLINE_FUNCTION
-  void operator()( const size_type & i, maxloc_type & lmaxloc ) const {
-	mag_type val = IPT::norm (m_x(i));
-    if( val > lmaxloc.val ) { lmaxloc.val = val; lmaxloc.loc = i; }
+  KOKKOS_INLINE_FUNCTION void
```

Comment:
You should be able to use `Kokkos::MaxLoc` to replace `init` and `join` here.

Reply:
@mhoemmen I used ```Kokkos::MaxLoc``` in my previous commit. Since ```iamax``` returns only the index, I found that ```Kokkos::MaxLoc``` works well for the 1-D input case when the index result returned to host. But, when the index result returned to a 0-D view on device, I need an extra copy the ```Kokkos::MaxLoc```'s ```loc``` result from host to device.
A custom reduction with the index as the reduction value can help in this case.

Reply:
Sure, that's ok -- thanks for thinking about it, though.